src/mozillavpn.cpp

```diff
@@ -1290,6 +1292,7 @@ void MozillaVPN::hardReset() {
 void MozillaVPN::hardResetAndQuit() {
   logger.debug() << "Hard reset and quit";
   hardReset();
```

Comment:
We start the app in`StateInitialize`, right? If that's the case, wouldn't connecting `StateInitialize` to removing the config end up with some bad side effects? (We could mitigate by only destroying it when we *move into* StateInitialize from another state, but that seems like an antipattern, and is fragile if we ever change the app to start in StatePreInit (and then go into StateInitialize or something like that).

Reply:
Good call. Should we also add it to `MozillaVPN::reset`? I don't see a connection between `hardReset` and `reset`, though I may be missing it.

Reply:
I've added it to `hardReset`

Reply:
I'd say so, since thats the codepath that is taken when a device is removed (providing we can't find a one size fits all solution as mentioned in the comments below)

Reply:
I think I'm on board, but want to call this out - We have a few error paths that bounce someone back to re-authenticate. In these cases, they log back in and (as of now) they can turn on the VPN by flicking the switch. If we add this iOS system config removal to `reset`, they'll now need to re-add the config (w/ the system dialog box) when they re-activate the VPN after logging back in. On the other hand, it's more defensive code, and prevents future potential issues. Thoughts on those tradeoffs?